superset/assets/src/components/FilterableTable/FilterableTable.jsx

```diff
@@ -152,21 +153,32 @@ export default class FilterableTable extends PureComponent {
     const widthsByColumnKey = {};
     this.props.orderedColumnKeys.forEach((key) => {
       const colWidths = this.list
-        .map(d => getTextWidth(d[key]) + PADDING) // get width for each value for a key
-        .push(getTextWidth(key) + PADDING); // add width of column key to end of list
+        // get width for each value for a key
+        .map(d => getTextWidth(
+          this.getCellContent({ cellData: d[key], columnKey: key })) + PADDING,
+        )
+        // add width of column key to end of list
+        .push(getTextWidth(key) + PADDING);
       // set max width as value for key
       widthsByColumnKey[key] = Math.max(...colWidths);
     });
     return widthsByColumnKey;
   }
 
-  fitTableToWidthIfNeeded() {
-    const containerWidth = this.container.clientWidth;
-    if (this.totalTableWidth < containerWidth) {
-      // fit table width if content doesn't fill the width of the container
-      this.totalTableWidth = containerWidth;
+  getCellContent({ cellData, columnKey }) {
+    const complexColumn = this.props.expandedColumns.some(
+      name => name.startsWith(columnKey + '.'),
+    );
+    const content = String(cellData);
+    let type;
+    if (content.substring(0, 1) === '[') {
+      type = '[…]';
+    } else if (content.substring(0, 1) === '[') {
+      type = '{…}';
```

Comment:
When would we hit this?

Reply:
Ah, it was supposed to be `content.substring(0, 1) === '{'`. Will fix.

Reply:
Unfortunately, the content doesn't distinguish between rows and arrays. A row content will also be "[ "a", "b", ... ]".

Reply:
I understand, but I think it's fine to show the rows as `[...]`, since we represent the data as Javascript arrays.

Reply:
Sounds good.